deploy/docs/Security_Best_Practices.md

```diff
@@ -2,6 +2,59 @@
 
 - [Hardening fluentd StatefulSet with `securityContext`](#hardening-fluentd-statefulset-with-securitycontext)
 
+## Minimal required capabilities
+
+This section explains the basic requirements for the collection solution in terms of network and security policies.
+It is intended to serve as a guide for running collection in highly locked-down environments.
+
+Pipelines for the three different data types collection supports - logs, metrics and traces - are completely
+independent in this respect, so we're going to discuss each one separately. If a particular data type is
+not enabled in the configuration, there is no need to do anything, as the respective components will
+simply not be started.
+
+Logically, the pipeline for a data type is composed of the following components
+
+```
+Collector -> (Optional) Metadata Enrichment -> Sumologic Backend
+```
+
+Different applications may serve the same role here - for example, FluentD is currently used for metadata enrichment
+for both logs and metrics, but the plan is to replace it with the Opentelemetry Collector in the near future. Nonetheless,
+the required capabilities are only based on the role, not the specific application.
+
+### Logs
+
+Log collection is done by a node agent, which reads them directly from the node by having the right directories
+\- `/var/log/containers` for container logs and `/var/log/journal` for journald logs - mounted as volumes in the
+agent container. In order to read them, the agent container needs to run as a user with the right permissions.
+
+As Kubernetes distributions have different ACLs set for these directories, Fluent-Bit runs as the root user by default
+for maximum out-of-the-box compatibility. This can be set to any user or group id with permission to read log files
+in the aforementioned directories on the node. In terms of collection configuration, the value to modify is
+`fluent-bit.securityContext` for Fluent-Bit.
+
+The log collector needs to be able to talk to the log metadata enrichment service. It does not need to do any other
+requests over the network.
+
+The log metadata enrichment service needs to be able to talk to the Sumo Logic backend receiver endpoints. It also needs
+to be able to access the Kubernetes API Server to pull metadata.
```

Comment:
> to pull metadata

Or to get notified via watch when user is using otelcol for metadata enrichment.

Reply:
This is intended as a high-level description. It pulls the data via a WATCH.

Reply:
It doesn't technically pull it. The notifications get pushed to the client AFAIK.

Reply:
obtain, receive, get 🤷

Reply:
I like `obtain`.